lib/Cake/View/Helper/HtmlHelper.php

```diff
@@ -180,8 +180,17 @@ public function __construct(View $View, $settings = array()) {
  * @see HtmlHelper::link() for details on $options that can be used.
  * @link http://book.cakephp.org/2.0/en/core-libraries/helpers/html.html#creating-breadcrumb-trails-with-htmlhelper
  */
-	public function addCrumb($name, $link = null, $options = null) {
-		$this->_crumbs[] = array($name, $link, $options);
+	public function addCrumb($name, $link = null, $options = array()) {
```

Comment:
Shouldn't the typehint also stay `= null`?
And you need to check for is_array() as also a string can be passed.